src/subscriptions/subscriptionBackfill.ts

```diff
@@ -129,15 +135,22 @@ export function makeBackfiller(jsonRpcSenders: JsonRpcSenders) {
     if (fromBlockInclusive >= toBlockExclusive) {
       return [];
     }
-    const batchParts: BatchPart[] = [];
+    let batchParts: BatchPart[] = [];
+    const allHeads: NewHeadsEvent[] = [];
     for (let i = fromBlockInclusive; i < toBlockExclusive; i++) {
       batchParts.push({
         method: "eth_getBlockByNumber",
         params: [toHex(i), false],
       });
+      if (batchParts.length % MAX_BATCH_SIZE === 0) {
+        const heads = await jsonRpcSenders.sendBatch(batchParts);
+        allHeads.concat(heads.map(toNewHeadsEvent));
+        batchParts = [];
```

Comment:
Possible to send all batches concurrently async? This might cause increased latency.